### PR TITLE
First Draft: Remove BackgroundMapSettings' provider name and type

### PR DESCRIPTION
**Pull request by mdastous-bentley**
First draft to remove the provider name/type from BackgroundMapSettings and stop syncing settings between BackgroundMapSettings and BaseMapProps.

The following sample API call (only true for changing the provider settings).
**displayStyle.changeBackgroundMapProps**({providerName: "BingProvider", providerData: {mapType:BackgroundMapType.Hybrid}});
would be replaced by 
**displayStyle.changeBaseMapContentProps**({providerName: "BingProvider", providerData: {mapType:BackgroundMapType.Hybrid}});

Also any previous calls to:
**displayStyle.changeBaseMapProps**
should be replace with:
**displayStylechangeBaseMapContentProps**

A decision is needed about the remaining settings left in BackgroundMapsProps, should it migrated under the BaseMapProps or remains in BackgroundMapsProps.

### CODE REVIEW

File: core/common/src/MapImagerySettings.ts [hunk 1]
```diff
@@ -22,31 +22,214 @@ export type BaseLayerProps = MapLayerProps | ColorDefProps;
...
+    return props?.hasOwnProperty("providerName");
+  }
 }
 
 /** Normalized representation of base layer properties -- these can be represented by either a full map layer or a simple color.
  * @beta
  */
-export type BaseLayerSettings = MapLayerSettings | ColorDef;
+export type BaseLayerContent = MapLayerSettings | ColorDef | BackgroundMapProvider;
+export type BaseLayerContentProps = MapLayerProps | ColorDefProps | BackgroundMapProviderProps;
```
Comment (mdastous-bentley) [inline core/common/src/MapImagerySettings.ts:155]: New 'content' type that specify the base layer as either a Map layer definition, a color or a BackgroundMapProvider

File: core/common/src/MapImagerySettings.ts [hunk 2]
```diff
@@ -22,31 +22,214 @@ export type BaseLayerProps = MapLayerProps | ColorDefProps;
...
+
+export interface BaseLayerSettingsProps {
```
Comment (mdastous-bentley) [inline core/common/src/MapImagerySettings.ts:157]: BaseLayerSettings is now composed of two parts: content (similar to the previous definition) and displaySettings that would replace the actual BackgroundMapsSettings if we decide to group everything under the BaseLayer/BaseMap

File: core/common/src/DisplayStyleSettings.ts [hunk 3]
```diff
@@ -546,7 +548,10 @@ export class DisplayStyleSettings {
     this._monochromeMode = MonochromeMode.Flat === this._json.monochromeMode ? MonochromeMode.Flat : MonochromeMode.Scaled;
 
     this._backgroundMap = BackgroundMapSettings.fromJSON(this._json.backgroundMap);
-    this._mapImagery = MapImagerySettings.fromJSON(this._json.mapImagery, this._json.backgroundMap);
+
+    // TODO
+    // Should we have a migration path here with the old JSON
+    this._mapImagery = MapImagerySettings.fromJSON(this._json.mapImagery);
```
Comment (mdastous-bentley) [inline core/common/src/DisplayStyleSettings.ts:554]: How de we want to handle to migration from previous settings format?

Comment (markschlosseratbentley) [reply core/common/src/DisplayStyleSettings.ts:554]: Perhaps a new method that does the conversion, if desired?

File: core/common/src/MapImagerySettings.ts (hunk 2 above)
Comment (markschlosseratbentley) [reply core/common/src/MapImagerySettings.ts:157]: Grouping everything one place could perhaps avoid confusion.

**Comment by pmconne** [comment]
This PR needs to:
- Continue to respect legacy BackgroundMapProps persisted in saved views (whether saved in the iModel or the project settings service).
- Make it trivial for existing callers to upgrade their code. IMO changing `changeBackgroundMapProps` to `changeBaseMapContentProps` accepting the exact same arguments is not useful.

**Pull request closed without merging**